Detectors/TPC/simulation/include/TPCSimulation/Point.h

```diff
@@ -4,13 +4,59 @@
 #define ALICEO2_TPC_POINT_H
 
 #include "SimulationDataFormat/BaseHits.h"
+#include <vector>
 
 namespace o2 {
 namespace TPC {
 
+// a minimal and plain TPC hit class
+class ElementalHit {
+ public:
+  Point3D<float> mPos; // cartesian position of Hit
+  float mTime = -1;    // time of flight
+  float mELoss = -2;   // energy loss
+
+ public:
+  ElementalHit() = default; // for ROOT IO
+  ~ElementalHit() = default;
+
+  // constructor
+  ElementalHit(float x, float y, float z, float time, float e)
+    :  mPos(x, y, z), mTime(time), mELoss(e) {}
+
+  ClassDefNV(ElementalHit,1);
+};
+
+// a higher order hit class encapsulating
+// a set of elemental hits belonging to the same trackid (and sector)
+// construct used to do less MC truth linking and to save memory
+// this hitcontainer is linkable with FairLinks,
+// and can be stored as element of a TClonesArray into a branch
+class LinkableHitGroup : public o2::BaseHit {
+public:
+  LinkableHitGroup() : mHits() {}
+
+  LinkableHitGroup(int trackID) : mHits() {
+    SetTrackID(trackID);
+  }
+
```

Comment:
No ... since the stack still wants to modify this field a posterio. At some moment we could get rid of the trackID field in the base class but is for later (we have discussed this elsewhere).